service/src/registry.ts

```diff
@@ -735,6 +739,36 @@ export class Registry {
     /*Codes_SRS_NODE_IOTHUB_REGISTRY_05_013: [The fromSharedAccessSignature method shall return a new instance of the `Registry` object.]*/
     return new Registry(config);
   }
+
+  private _normalizeAuthentication(deviceInfo: Registry.DeviceDescription): void {
+    if (!deviceInfo.hasOwnProperty('authentication')) {
+      /* Codes_SRS_NODE_IOTHUB_REGISTRY_06_028: [A device information with no authentication will be normalize with the following authentication:
+      authentication : {
+        type: 'sas',
+        symmetricKey: {
+          primaryKey: '',
+          secondaryKey: ''
+        }
+      }
+      ] */
+      deviceInfo.authentication = {
+          type: 'sas',
+          symmetricKey: {
+            primaryKey: '',
+            secondaryKey: ''
+          }
+        };
+    /* Codes_SRS_NODE_IOTHUB_REGISTRY_06_029: [** A device information with an authentication object that contains a `type` property is considered normalized.] */
+    } else if (!deviceInfo.authentication.hasOwnProperty('type')) {
```

Comment:
please leave an explanation as to why this is a better design than the requested change. Being dismissive is not gonna help your case.